File: DiscordBot/src/main/java/com/example/demo/Listeners/implementations/DeleteReactionListenerImpl.java
package com.example.demo.Listeners.implementations;

import com.example.demo.Listeners.DeleteReactionListener;
import org.javacord.api.event.message.reaction.ReactionAddEvent;
import org.springframework.stereotype.Component;

@Component
public class DeleteReactionListenerImpl implements DeleteReactionListener {
    @Override
    public void onReactionAdd(ReactionAddEvent event) {
        if (event.getEmoji().equalsEmoji("\uD83D\uDC4E")) {
            event.deleteMessage();
        }
    }
}

File: KotlinBot/src/main/kotlin/com/example/KotlinBot/Listener/Implementation/DeleteReactionListenerImpl.kt
package com.example.KotlinBot.Listener.Implementation

import com.example.KotlinBot.Listener.DeleteReactionListener
import org.javacord.api.event.message.reaction.ReactionAddEvent
import org.springframework.context.annotation.Primary
import org.springframework.stereotype.Component

@Component
@Primary
class DeleteReactionListenerImpl : DeleteReactionListener {
    override fun onReactionAdd(event: ReactionAddEvent?) {
        if (event!!.emoji.equalsEmoji("\uD83D\uDC4E")) {
            event.deleteMessage()
        }
    }

}